templates/web_gateway/web_gateway.py

```diff
@@ -28,6 +29,26 @@ def __init__(self, name=None, guid=None, data=None):
         self._etcds_name = 'etcds_%s' % self.guid
         self._coredns_name = "coredns_%s" % self.guid
         self._traefik_name = "traefik_%s" % self.guid
+        self.recurring_action('_monitor', 30)  # every 30 seconds
+
+    def _monitor(self):
+        self.logger.info('Monitor web gateway %s' % self.name)
+        self.state.check('actions', 'start', 'ok')
+        try:
+            self._etcd_cluster.state.check('status','running','ok')
+            self._traefik.state.check('status','running','ok')
+            self._coredns.state.check('status','running','ok')
+            self.state.set('status', 'running', 'ok')
+        except StateCheckError:
+            self.state.delete('status', 'running')
+
+        cluster_connection = self._etcd_cluster.schedule_action('connection_info').wait(die=True).result
+        if self.data['etcdConnectionInfo']['etcds'] != cluster_connection['etcds']:
+            self.data['etcdConnectionInfo']['etcds'] = cluster_connection['etcds']
+            traefik_endpoint = ','.join(['{}:{}'.format(connection['ip'], connection['client_port']) for connection in self.data['etcdConnectionInfo']['etcds']])
```

Comment:
May be move the traefik endpoint and coredns endpoint to a function instead of having it twice, here and in `install` action. And then you can just do: `self._coredns.schedule_action('update_endpoint', args=self._coredns_endpoint()).wait(die=True)`